### PR TITLE
Get base URL from `LIVEBLOCKS_BASE_URL` envvar if not explicitly defined

### PR DESCRIPTION
**Pull request by nvie**
This will make it much easier to test Liveblocks against a locally running backend, i.e. by running an example instead of:

```bash
$ turbo run dev
```

…and instead just run the following command without making any config/code changes:

```bash
$ env NEXT_PUBLIC_LIVEBLOCKS_BASE_URL=http://127.0.0.1:3333 turbo run dev
```

Because this requires both the frontend (in `createClient()`) and the backend (in `new Liveblocks()`) to read the same base URL value, I've stayed pragmatic and allowed it to read the `NEXT_PUBLIC_` value (and `VITE_`) here, since in a Next.js setup it will otherwise not be visible to the frontend for obvious reasons.

For non-Next.js projects, it will also be read from `LIVEBLOCKS_BASE_URL` too.

### CODE REVIEW

File: packages/liveblocks-core/src/client.ts [hunk 1]
```diff
@@ -289,15 +289,16 @@ export type ClientOptions<TUserMeta extends BaseUserMeta = BaseUserMeta> = {
...
 //
 
-function getBaseUrlFromClientOptions(clientOptions: ClientOptions) {
-  if ("liveblocksServer" in clientOptions) {
-    throw new Error("Client option no longer supported");
-  }
+function getBaseUrl(baseUrl?: string | undefined): string {
+  baseUrl ||=
+    process.env.LIVEBLOCKS_BASE_URL ||
+    process.env.NEXT_PUBLIC_LIVEBLOCKS_BASE_URL ||
```
Comment (marcbouchenoire) [inline packages/liveblocks-core/src/client.ts:295]: We could also add `VITE_LIVEBLOCKS_BASE_URL` to be exhaustive and cover the remaining examples (Vite covers non-Next.js React examples but also Vue, Svelte, etc).

File: packages/liveblocks-node/src/utils.ts [hunk 2]
```diff
@@ -1,6 +1,21 @@
 import type { Brand } from "@liveblocks/core";
 
-export const DEFAULT_BASE_URL = "https://api.liveblocks.io";
+const DEFAULT_BASE_URL = "https://api.liveblocks.io";
+
+export function getBaseUrl(baseUrl?: string | undefined): string {
+  baseUrl ||=
+    process.env.LIVEBLOCKS_BASE_URL ||
+    process.env.NEXT_PUBLIC_LIVEBLOCKS_BASE_URL ||
```
Comment (marcbouchenoire) [inline packages/liveblocks-node/src/utils.ts:8]: Ditto, regarding `VITE_LIVEBLOCKS_BASE_URL`.

**Review by marcbouchenoire — APPROVED** [review]
Great improvement. 👌

File: packages/liveblocks-core/src/client.ts (hunk 1 above)
Comment (nvie) [reply packages/liveblocks-core/src/client.ts:295]: Great point. Added in ad2e1c5261aabd652056cdad1286cf097886797d 👍

**Pull request merged**